python/res/fm/ecl/ecl_run.py

```diff
@@ -0,0 +1,385 @@
+       from res.fm.ecl import EclRun
+
+       run = EclRun(sys.argv)
+       run.runEclipse( )
+
+
+    Name of executable
+    ==================
+
+    Observe that the EclRun.__init__() initializer determines which simulator
+    to use by checking the value of sys.argv[0] - i.e. typically the name of
+    the executable. The currently recognized names are:
+
+        run_ecl100 : Run ECLIPSE100
+        run_ecl300 : Run ECLIPSE300
+        run_flow   : Run OPM/flow
+
+    The names 'ecl100', 'ecl300' and 'flow' will also be used as keys to look
+    up information about the simulator in the EclConfig class.
+
+    """
+    Eclipse100 = "run_ecl100"
+    Eclipse300 = "run_ecl300"
+    Eclipse100_nocheck = "run_ecl100_nocheck"
+    Flow = "run_flow"
+
+
+    def __init__(self , argv):
+        if 3 <= len(argv) <= 4:
+            self.check_status = True
+            config = EclConfig()
+
+            # First argument - name of simulator
+            script_name = os.path.basename( argv[0] )
+            if script_name == EclRun.Eclipse100:
+                simulator = "ecl100"
+            elif script_name == EclRun.Eclipse300:
+                simulator = "ecl300"
+            elif script_name == EclRun.Eclipse100_nocheck:
+                simulator = "ecl100"
+                self.check_status = False
+            elif script_name == EclRun.Flow:
+                simulator = "flow"
+            else:
+                raise ValueError("The name of the script:%s is invalid - must be run_ecl100 / run_ecl300" % script_name)
+
+
+            # Second argument
+            self.version = argv[1]
+
+
+            # Third argument - datafile, with optional leading path component
+            input_arg = argv[2]
+            (base , ext) = os.path.splitext( input_arg )
+            if ext and ext in [".data", ".DATA"]:
+                data_file = input_arg
+            else:
+                if input_arg.islower():
+                    data_file = input_arg + ".data"
+                else:
+                    data_file = input_arg + ".DATA"
+
+
+            if not os.path.isfile( data_file ):
+                raise IOError("No such file: %s" % data_file )
+            else:
+                (self.run_path , self.data_file)  = os.path.split( data_file )
+                (self.base_name , ext ) = os.path.splitext( self.data_file )
+
+                if self.run_path is None:
+                    self.run_path = os.getcwd()
+                else:
+                    self.run_path = os.path.abspath( self.run_path )
+
+            # Fourth argument - the number of CPU's - defaults to one if not supplied.
+            if len(argv) == 3:
+                self.num_cpu = 1
+                self.sim = config.sim( simulator, self.version )
+            else:
+                self.num_cpu = int(argv[3])
+                self.sim = config.mpi_sim( simulator, self.version)
+
+        else:
+            raise ValueError("Must have three/four arguments")
+
+
```

Comment:
I have only quickly skimmed through the rest of EclRun, as I guess it was not significantly modified wrt ert-statoil (and it also includes a good deal of implementation details I am not too familiar with). If you'd like me to check this in more detail let me know.

Reply:
That is fine - it has worked for several years.